### ISSUE TITLE
G4 Doorbell Pro Package Camera

### ISSUE THREAD

**Issue by HeavneMa** [hjdhjd/homebridge-unifi-protect#1152]
### Homebridge UniFi Protect Version

7.8.2

### Homebridge Platform and OS

Raspberry Pi 4B+ OS Lite 64-bit kernel 6.6 Debian 12

### Homebridge Version

1.8.5

### Node Version

20.18.0

### UniFi OS Version

4.1.19

### UniFi Protect Controller Version

5.1.57

### Describe the problem

Package Camera works at first config, then reverts to showing the same as main doorbell camera. Have restarted HomeBridge, have restarted G4 Doorbell Pro, and reset / re-adopted G4 Doorbell Pro in Protect. It always reverts to showing the main cam in HomeKit after some time. Protect App and web page still work as designed.

### Homebridge HBUP JSON configuration

```json
{
    "controllers": [
        {
            "address": "cloudkey.doodle.home",
            "password": "*m*a*s*k*e*d",
            "username": "homebridge"
        }
    ],
    "name": "UniFi Protect",
    "options": [
        "Enable.Video.Transcode.Hardware.D8B3707C0211",
        "Enable.Video.HKSV.Record.Only.High.70A7415F6E01",
        "Enable.Video.HKSV.Record.Only.High.28704E126B19",
        "Enable.Video.HKSV.Record.Only.High.F4E2C672EC24",
        "Enable.Video.Stream.Only.High.F4E2C672EC24",
        "Enable.Video.Stream.Only.High.28704E126B19",
        "Enable.Video.Stream.Only.High.70A7415F6E01",
        "Enable.Device.Standalone.F4E2C672EC24",
        "Enable.Device.Standalone.28704E126B19",
        "Enable.Device.Standalone.70A7415F6E01",
        "Enable.Video.Stream.Only.High.F4E2C6DB4EF9"
    ],
    "_bridge": {
        "username": "0E:52:B2:60:19:8B",
        "port": 54736
    },
    "platform": "UniFi Protect"
}
```

### Relevant log output

```shell
[04/12/2024, 14:23:09] [UniFi Protect] Initializing UniFi Protect platform...
[04/12/2024, 14:23:09] [UniFi Protect] Initializing child bridge 0E:52:B2:60:19:8B
[04/12/2024, 14:23:10] [UniFi Protect] Launched child bridge with PID 11315
[04/12/2024, 14:23:11] [UniFi Protect] Loaded homebridge-unifi-protect v7.8.2 child bridge successfully
[04/12/2024, 14:23:11] Loaded 1 cached accessories from cachedAccessories.0E52B260198B.
[04/12/2024, 14:23:11] Homebridge v1.8.5 (HAP v0.12.3) (Nest Protect) is running on port 35455.
[04/12/2024, 14:23:11] 

NOTICE TO USERS AND PLUGIN DEVELOPERS
> Homebridge 2.0 is on the way and brings some breaking changes to existing plugins.
> Please visit the following link to learn more about the changes and how to prepare:
> https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0

[04/12/2024, 14:23:11] [UniFi Protect] Using FFmpeg version: 7.0.2-homebridge-alpine-aarch64-static.
[04/12/2024, 14:23:11] Homebridge v1.8.5 (HAP v0.12.3) (UniFi Protect) is running on port 54736.
[04/12/2024, 14:23:11] 

NOTICE TO USERS AND PLUGIN DEVELOPERS
> Homebridge 2.0 is on the way and brings some breaking changes to existing plugins.
> Please visit the following link to learn more about the changes and how to prepare:
> https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0

[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Connected to cloudkey.doodle.home (UniFi Protect 5.1.57 running on UniFi OS 4.1.9).
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Discovered nvr: Haus of Doodle [UCK G2 Plus] (address: 192.168.1.10 mac: F4E2C6DB4EF9).
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Discovered camera: Doodle Doorbell [G4 Doorbell Pro] (address: 192.168.1.239 mac: F4E2C672EC24).
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Discovered camera: Living Area Cam [G4 Instant] (address: 192.168.1.91 mac: 28704E126B19).
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Discovered camera: Patio Cam [G4 Instant] (address: 192.168.1.160 mac: 70A7415F6E01).
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Doodle Doorbell [G4 Doorbell Pro]: Adding camera to HomeKit as a standalone device.
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Living Area Cam [G4 Instant]: Adding camera to HomeKit as a standalone device.
[04/12/2024, 14:23:13] [UniFi Protect] Haus of Doodle [UCK-G2-PLUS]: Patio Cam [G4 Instant]: Adding camera to HomeKit as a standalone device.
[04/12/2024, 14:23:13] Doodle Doorbell 53D4 is running on port 45175.
[04/12/2024, 14:23:13] Please add [Doodle Doorbell 53D4] manually in Home app. Setup Code: 450-11-146
[04/12/2024, 14:23:13] Living Area Cam 772D is running on port 44447.
[04/12/2024, 14:23:13] Please add [Living Area Cam 772D] manually in Home app. Setup Code: 450-11-146
[04/12/2024, 14:23:13] Patio Cam ABA0 is running on port 35539.
[04/12/2024, 14:23:13] Please add [Patio Cam ABA0] manually in Home app. Setup Code: 450-11-146
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: Video streaming configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: HomeKit Secure Video event recording configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: Enabled doorbell message switch: LEAVE PACKAGE AT DOOR.
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: Enabled doorbell message switch: DO NOT DISTURB.
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: Enabled doorbell message switch: Welcome to Haus of Doodle.
[04/12/2024, 14:23:13] [UniFi Protect] Patio Cam [G4 Instant]: Video streaming configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Patio Cam [G4 Instant]: HomeKit Secure Video event recording configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Living Area Cam [G4 Instant]: Video streaming configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Living Area Cam [G4 Instant]: HomeKit Secure Video event recording configured to use only: High.
[04/12/2024, 14:23:13] [UniFi Protect] Patio Cam [G4 Instant]: HKSV: hardware-accelerated 2688x1512@30fps (High) [HEVC], 2,000 kbps (10 second timeshift buffer).
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell [G4 Doorbell Pro]: HKSV: hardware-accelerated 1600x1200@30fps (High) [HEVC], 2,000 kbps (10 second timeshift buffer).
[04/12/2024, 14:23:13] [UniFi Protect] Doodle Doorbell Package Camera [G4 Doorbell Pro]: HKSV: hardware-accelerated 1600x1200@2fps (Package Camera) [HEVC], 500 kbps (10 second timeshift buffer).
[04/12/2024, 14:23:24] [UniFi Protect] Doodle Doorbell Package Camera [G4 Doorbell Pro]: Streaming request from 192.168.6.14: 640x480@30fps, 164 kbps. Hardware accelerated transcoding 1600x1200@2fps (Package Camera) [HEVC], 600 kbps [API].
[04/12/2024, 14:23:35] [UniFi Protect] Doodle Doorbell Package Camera [G4 Doorbell Pro]: Stopped video streaming session.
[04/12/2024, 14:30:34] [UniFi Protect] Doodle Doorbell Package Camera [G4 Doorbell Pro]: Streaming request from 192.168.6.14: 640x480@30fps, 164 kbps. Hardware accelerated transcoding 1600x1200@2fps (Package Camera) [HEVC], 600 kbps [API].
[04/12/2024, 14:30:42] [UniFi Protect] Doodle Doorbell Package Camera [G4 Doorbell Pro]: Stopped video streaming session.
```

### Acknowledgment that you are only running UniFi OS and UniFi Protect releases from the Ubiquiti Official release channel

- [X] I acknowledge that I am not running any early access / beta firmwares or operating systems from either Ubiquiti or Apple.

**Comment by HeavneMa** [comment]
Note - this was when I had the Doorbell set as a device to be manually added to HomeKit. The Package Camera would still add automatically when I added the HBUP bridge, but then I would manually add the main camera. I've cleared the cache of all devices, removed the bridge and manually added camera (including my G4 Instant) from HomeKit, set the Doorbell Camera to be added automatically with the bridge, and then re-started Homebridge completely - and re-added the HBUP bridge to HomeKit (automatically adding the main doorbell camera and package camera). Will update if the issue does not re-occur.
